### PR TITLE
[JIT] Support for ParameterDict getattr

### LINKED ISSUE #76602: `ParameterDict` doesn't work in TorchScript
### 🐛 Describe the bug

```
In [1]: import torch

In [2]: torch.__version__
Out[2]: '1.12.0.dev20220412'

In [3]: import torch as th
   ...: 
   ...: 
   ...: class TestMod(th.nn.Module):
   ...:     def __init__(self):
   ...:         super().__init__()
   ...:         self.params = th.nn.ParameterDict(
   ...:             {"hi": th.nn.Parameter(th.zeros(3)), "bye": th.nn.Parameter(th.zeros(3))}
   ...:         )
   ...: 
   ...:     def forward(self, x):
   ...:         return x + self.params["hi"] + self.params["bye"]
   ...: 
   ...: 
   ...: mod = TestMod()
   ...: smod = th.jit.script(mod)
```
gives
```
RuntimeError: 
Only ModuleList, Sequential, and ModuleDict modules are subscriptable:
  File "<ipython-input-3-6fafe8d819b8>", line 12
    def forward(self, x):
        return x + self.params["hi"] + self.params["bye"]
                   ~~~~~~~~~~~~~~~~ <--- HERE
```

See https://github.com/pytorch/pytorch/issues/25367.

### Versions

Occurs also on 1.10 stable.

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #77143

Adds support for scripting ParameterDicts and getattr() on them. It does
not support iterating on ParameterDicts because torch/nn/container.py
implementation of ParameterDict.items() uses a generator, which is not
supported by torchscript. torch/nn/container.py would need to be updated
so that iter gets correctly registered in python_sugared_value.cpp

Added a test in test_module_containers.py

### CODE REVIEW

File: torch/csrc/jit/python/python_sugared_value.cpp [hunk 1]
```diff
@@ -506,6 +515,33 @@ std::shared_ptr<SugaredDict> ModuleValue::getSugaredDict(
...
+    }
+  }
+
+  std::vector<SugaredValuePtr> keys;
+  std::vector<SugaredValuePtr> values;
+  for (const auto& name : paramNames) {
+    auto name_v =
+        std::make_shared<SimpleValue>(insertConstant(*m.graph(), name));
+    m.graph()->insertGetAttr(self_, name);
+    values.push_back(tryGetAttr(loc, m, name));
```
Comment (eellison) [inline torch/csrc/jit/python/python_sugared_value.cpp:535]: do we need to test if `tryGetAttr` is successful?

**Review by eellison — APPROVED** [review]
Nice, LGTM !!

File: torch/csrc/jit/python/python_sugared_value.cpp (hunk 1 above)
Comment (davidberard98) [reply torch/csrc/jit/python/python_sugared_value.cpp]: good idea, I added one

**Pull request closed without merging**